rectools/models/nn/item_net.py

```diff
@@ -0,0 +1,411 @@
+#  Copyright 2025 MTS (Mobile Telesystems)
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import typing as tp
+import warnings
+
+import torch
+import typing_extensions as tpe
+from torch import nn
+
+from rectools.dataset.dataset import Dataset, DatasetSchema
+from rectools.dataset.features import SparseFeatures
+
+
+class ItemNetBase(nn.Module):
```

Comment:
In the future we might actually use this for user nets (just like in DSSM, but much more flexible).
But I kinda like `ItemNet` naming - it's short and clear.

Possible naming is `EntityNet` but a bit more complicated

Reply:
Ok, let's keep as is for now